FW/FreeRTOS/Queue.hpp

```diff
@@ -0,0 +1,88 @@
+#ifndef _FREERTOS_QUEUE_HPP
+#define _FREERTOS_QUEUE_HPP
+
+#include "FreeRTOS.h"
+#include "queue.h"
+
+
+namespace FreeRTOS
+{
+    template<class T, size_t size>
+    class Queue {
+    public:
+        Queue() {
+            //m_handle = xQueueCreate(size, sizeof(T));
+            m_handle = xQueueCreateStatic(size, sizeof(T), m_buffer, &m_staticQueue);
+        }
+
+        ~Queue() {
+            vQueueDelete(m_handle);
+        }
+
+
+        /**
+         * @brief Add item to queue, if queue is full, thread will be paused until there
+         *        is room or timeout elapses. Do not use this within an interrupt, use
+         *        EnqueueFromInterrupt() instead. Will return true if item successfully
+         *        was added to queue, or false if timed out
```

Comment:
How about the ownership of item, is the data pointed to copied into the queue or does the queue take ownership of the data? Good to note in the doxygen, to avoid confusion and rummaging through the FreeRTOS docs. Same thing is good to note in the EnqueueFromInterrupt() and Dequeue() methods.

Reply:
No I don't mean there's something wrong, I just meant it might be good to note in the doxygen comment that the data pointed to will be copied. So FreeRTOS noobs like me don't need to go searching for the FreeRTOS docs to see if the message passing takes ownership of the data or simply copies it when we want to pass a message. But maybe that's universally understood when the code can run without an allocator.